bundles/best_practices.rst

```diff
@@ -166,6 +166,73 @@ the ``Tests/`` directory. Tests should follow the following principles:
    A test suite must not contain ``AllTests.php`` scripts, but must rely on the
    existence of a ``phpunit.xml.dist`` file.
 
+Travis
+------
+
+A popular way to test open source bundles is by using `Travis CI`_. A good practice
+is to support at least the two latest LTS versions of Symfony. One should also test
+test latest beta release. Here is a recommended configuration file (``.travis.yml``).
+
+.. code-block:: yaml
+
+    language: php
+    sudo: false
+    cache:
+        directories:
+            - $HOME/.composer/cache/files
+            - $HOME/symfony-bridge/.phpunit
+
+    env:
+        global:
+            - PHPUNIT_FLAGS="-v"
+            - SYMFONY_PHPUNIT_DIR="$HOME/symfony-bridge/.phpunit"
+
+    matrix:
+        fast_finish: true
+        include:
+              # Minimum supported Symfony version with the latest PHP version
+            - php: 7.2
+              env: COMPOSER_FLAGS="--prefer-stable --prefer-lowest" SYMFONY_DEPRECATIONS_HELPER="weak"
+
+              # Test the latest stable release
+            - php: 7.0
+            - php: 7.1
+            - php: 7.2
+              env: COVERAGE=true PHPUNIT_FLAGS="-v --coverage-text"
+
+              # Test LTS versions. This makes sure we do not use Symfony packages with version greater
+              # than 2 or 3 respectively. Read more at https://github.com/symfony/lts
+            - php: 7.2
+              env: DEPENDENCIES="symfony/lts:^2"
+            - php: 7.2
+              env: DEPENDENCIES="symfony/lts:^3"
+
+              # Latest commit to master
+            - php: 7.2
+              env: STABILITY="dev"
+
+        allow_failures:
+              # Dev-master is allowed to fail.
+            - env: STABILITY="dev"
+
+    before_install:
+        - if [[ $COVERAGE != true ]]; then phpenv config-rm xdebug.ini || true; fi
+        - if ! [ -z "$STABILITY" ]; then composer config minimum-stability ${STABILITY}; fi;
+        - if ! [ -v "$DEPENDENCIES" ]; then composer require --no-update ${DEPENDENCIES}; fi;
+
+    install:
+        # To be removed when this issue will be resolved: https://github.com/composer/composer/issues/5355
+        - if [[ "$COMPOSER_FLAGS" == *"--prefer-lowest"* ]]; then composer update --prefer-dist --no-interaction --prefer-stable --quiet; fi
+        - composer update ${COMPOSER_FLAGS} --prefer-dist --no-interaction
+        - ./vendor/bin/simple-phpunit install
+
+    script:
+        - composer validate --strict --no-check-lock
```

Comment:
whats the point of this? wouldn't composer update fail if the composer.json is invalid which is run before anyway?

Reply:
afaik the `--strict` makes this fail even if there are just warnings (e.g. composer.json is not suitable for publishing, alphabetical order of dependencies etc)